### PR TITLE
Add redirect callback and services

### PR DESCRIPTION
**Pull request by johangant**
Allows us to tie up some loose ends from the D7 migration using the built in redirect module, rather than bundle them into something cruder like nginx config or .htaccess

### CODE REVIEW

File: migrate_nidirect_node/src/EventSubscriber/PostMigrationSubscriber.php [hunk 1]
```diff
@@ -126,6 +139,37 @@ public function onMigratePostImport(MigrateImportEvent $event) {
...
+    $this->redirects();
+  }
+
+  /**
+   * Guarantee certain redirects exist after each migration.
+   */
+  protected function redirects() {
+    $redirects = [
+      'contacts/contacts-az/police-service-northern-ireland' => 'internal:/node/77',
+      'contacts/contacts-az/migrant-help' => 'internal:/node/166',
```
Comment (neilblair) [inline migrate_nidirect_node/src/EventSubscriber/PostMigrationSubscriber.php:153]: Johan will this be the mechanism to fix all links to contacts? Aside from these two examples, there are likely to be hundreds more ...

**Comment by neilblair** [comment]
@johangant Just have a query about whether this will work for fixing all potential links to contact pages that are likely to be broken because the alias pattern differs between D7 and D8

**Comment by johangant** [comment]
> @johangant Just have a query about whether this will work for fixing all potential links to contact pages that are likely to be broken because the alias pattern differs between D7 and D8

You're right, I've misread it. I'll close this off and rework it.

**Comment by johangant** [comment]
Re-opening this with a different angle of attack:

* Adding to the post migrate event handler, we can now check for aliases for all D7 contacts. Where we don't already have an alias or redirect for that node in D8, we create a redirect to the canonical path for it.
* We can run this many times over and over and it won't trip up on duplicate keys due to the checks as it goes. If we've already got all aliases... it skips over quickly to completion.

File: migrate_nidirect_node/src/EventSubscriber/PostMigrationSubscriber.php (hunk 1 above)
Comment (johangant) [reply migrate_nidirect_node/src/EventSubscriber/PostMigrationSubscriber.php]: Absolutely right, I thought it was just for two items for some reason. Latest commit adds a process to cover them all in one go.

**Review by omahm — APPROVED** [review]
Nice one, LGTM

**Pull request merged**